### PR TITLE
Fix replace ingress_lossless_pool

### PR DESCRIPTION
**Pull request by xincunli-sonic**
### Description of PR

Summary:
Fixes # (issue) ADO: 28886361

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [x] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?

This is an improvement for `replace` case in a [GCU E2E Testing](https://github.com/sonic-net/sonic-mgmt/pull/11739)

#### How did you do it?
If the op is replace, then just check expected failure, no need do further steps.

### CODE REVIEW

File: tests/generic_config_updater/test_incremental_qos.py [hunk 1]
```diff
@@ -241,11 +241,15 @@ def test_incremental_qos_config_updates(duthost, tbinfo, ensure_dut_readiness, c
         output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
         if op == "replace" and not field_value:
             expect_op_failure(output)
-
-        if is_valid_platform_and_version(duthost, "BUFFER_POOL", "Shared/headroom pool size changes", op, field_value):
-            expect_op_success(duthost, output)
-            ensure_application_of_updated_config(duthost, configdb_field, value)
         else:
-            expect_op_failure(output)
+            if is_valid_platform_and_version(duthost,
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_incremental_qos.py:245]: minor comment: use `elif`

**Comment by qiluo-msft** [comment]
The PR description is empty? Are you fixing an github issue?

File: tests/generic_config_updater/test_incremental_qos.py (hunk 1 above)
Comment (xincunli-sonic) [reply tests/generic_config_updater/test_incremental_qos.py:245]: For non replace op, it requires validation, which is not a parallel check with op. So, for readability, I would prefer keep in current way.

**Comment by xincunli-sonic** [comment]
Added, this is addressing a nightly test issue.

---
In reply to: [2455961430](https://github.com/sonic-net/sonic-mgmt/pull/15332#issuecomment-2455961430) [](http://example.com/codeflow?ancestors=2455961430)

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please check with other reviewers.

File: tests/generic_config_updater/test_incremental_qos.py [hunk 2]
```diff
@@ -241,11 +241,15 @@ def test_incremental_qos_config_updates(duthost, tbinfo, ensure_dut_readiness, c
         output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
         if op == "replace" and not field_value:
             expect_op_failure(output)
```
Comment (kperumalbfn) [inline tests/generic_config_updater/test_incremental_qos.py:243]: For 'replace' and no buffer_pool in the config_db, could you add a log message that buffer_pool and the field value doesn't exist. Would be easy for debugging

File: tests/generic_config_updater/test_incremental_qos.py [hunk 3]
```diff
@@ -241,11 +241,15 @@ def test_incremental_qos_config_updates(duthost, tbinfo, ensure_dut_readiness, c
...
-            expect_op_failure(output)
+            if is_valid_platform_and_version(duthost,
+                                             "BUFFER_POOL",
+                                             "Shared/headroom pool size changes",
+                                             op,
+                                             field_value):
+                expect_op_success(duthost, output)
+                ensure_application_of_updated_config(duthost, configdb_field, value)
+            else:
+                expect_op_failure(output)
```
Comment (kperumalbfn) [inline tests/generic_config_updater/test_incremental_qos.py:253]: Error log for the failure?

**Comment by bingwang-ms** [comment]
Which test is failing without this fix?

File: tests/generic_config_updater/test_incremental_qos.py (hunk 3 above)
Comment (xincunli-sonic) [reply tests/generic_config_updater/test_incremental_qos.py:253]: Added in expect_op_failure

File: tests/generic_config_updater/test_incremental_qos.py (hunk 2 above)
Comment (xincunli-sonic) [reply tests/generic_config_updater/test_incremental_qos.py:243]: Added

**Review by kperumalbfn — APPROVED** [review]
LGTM

**Pull request merged**